frontend/src/app/AppInsights.js

```diff
@@ -5,20 +5,23 @@ import { createBrowserHistory } from "history";
 
 const browserHistory = createBrowserHistory({ basename: "" });
 const reactPlugin = new ReactPlugin();
-const appInsights = new ApplicationInsights({
-  config: {
-    instrumentationKey: process.env.REACT_APP_APPINSIGHTS_KEY,
-    extensions: [reactPlugin],
-    loggingLevelConsole: 2,
-    disableFetchTracking: false,
-    enableAutoRouteTracking: true,
-    loggingLevelTelemetry: 2,
-    maxBatchInterval: 0,
-    extensionConfig: {
-      [reactPlugin.identifier]: { history: browserHistory },
-    },
-  },
-});
-appInsights.loadAppInsights();
+const appInsights =
+  process.env.NODE_ENV === "production"
+    ? new ApplicationInsights({
+        config: {
+          instrumentationKey: process.env.REACT_APP_APPINSIGHTS_KEY,
+          extensions: [reactPlugin],
+          loggingLevelConsole: 2,
+          disableFetchTracking: false,
+          enableAutoRouteTracking: true,
+          loggingLevelTelemetry: 2,
+          maxBatchInterval: 0,
+          extensionConfig: {
+            [reactPlugin.identifier]: { history: browserHistory },
+          },
+        },
+      })
+    : undefined;
+appInsights?.loadAppInsights();
```

Comment:
Good question! This approach for disabling AppInsights in pre-prod environments is definitely _not_ great, and I think it should be improved if it's going to get committed to `main`

I'm going to leave this comment thread (and the related one below) open and come back to it later

Reply:
The Azure stuff definitely seems like it was written by Microsoft, it's pretty strange and not very JS-like. We could just do a better wrapper around the whole mess. In dev mode it would be nice to see console messages where we'd normally get telemetry, just to have an idea of the volume we're creating.

Reply:
Of course right now we get plenty of console messages since there's an error every time it tries to report. 😸